pkg/checksdb/check.go

```diff
@@ -74,14 +70,17 @@ func NewCheck(id string, labels []string) *Check {
 	return check
 }
 
-func (check *Check) Abort() {
+func (check *Check) Abort(reason string) {
 	check.mutex.Lock()
 	defer check.mutex.Unlock()
 
-	check.abortChan <- true
+	abortMsg := check.ID + " issued non-graceful abort: " + reason
+
+	check.abortChan <- abortMsg
+	panic(AbortPanicMsg(abortMsg))
```

Comment:
Why do we need to panic here?

Reply:
I believe this is to trigger the [recover()](https://gobyexample.com/recover) func.

Reply:
right, but what will the recover do in this case?
`			// Don't do anything in case the check was manually aborted by check.Abort().
			if msg, ok := r.(AbortPanicMsg); ok {
				log.Warn("Check was manually aborted, msg: %v", msg)
				err = fmt.Errorf("%v", msg)
				return
			}`
It seems that we are just going back to normal exec mode for a panic we created.
Won't just using the signal work the same?

Reply:
Normal panics from check functions, like index out-of-range runtime panics will make this check to be flagged as errored and the remaining checks of this group will be skipped, but the other group's checks will run normally: that was the original behaviour in ginkgo.

This is just a simple/quick replacement/alternate mechanism for ginkgo's AbortSuite(), so this specific panic recovery avoids running any other check of this group, while relying on the signal receiver to not run any other group's checks. Also, it relies on the signal receiver to call the OnAbort() to flag the current check as aborted and the remaining as skipped.

Anyways, I'll try to improve and simplify the whole mechanism in further PRs.